fix #54 updateShouldNotify (#57)
rrousselGit/state_notifier: packages/state_notifier/README.md

As such, instead of using `==`, [StateNotifier] relies on `identical` to compare
objects.  
This way, when using [StateNotifier] with simple states like `int`/enums, it will
correctly filter identical states. At the same time, this preserves performance
on complex states, at `identical` will not perform a deep object comparison.

> ### can i override this behavior ?

Reviewer: ```suggestion
### How to change this behavior?
```

> yes by overriding `updateShouldNotify(T old,T current)`

Reviewer: ```suggestion
You can override the method `updateShouldNotify(T old,T current)` of a `StateNotifier` to change the default behaviour, such as for:
- using `==` instead of `identical` to filter updates, for deep state comparison
- always return `true` to revert to older behaviors of `StateNotifier`
```

As such, instead of using `==`, [StateNotifier] relies on `identical` to compare
objects.  
This way, when using [StateNotifier] with simple states like `int`/enums, it will
correctly filter identical states. At the same time, this preserves performance
on complex states, at `identical` will not perform a deep object comparison.

### Using a custom notification filter logic

You can override the method `updateShouldNotify(T old,T current)` of a `StateNotifier` to change the default behaviour, such as for:
- using `==` instead of `identical` to filter updates, for deep state comparison
- always return `true` to revert to older behaviors of `StateNotifier`

```dart
  @override

>   bool updateShouldNotify(int old, int current) {
>     /// only update if the new state is not equal to the old state
>     return current != old;

Reviewer: ```suggestion
  bool updateShouldNotify(User old, User current) {
    /// only update the User content changes, even if using a different instance
    return old.name != current.name && old.age != current.age;
```
